packages/hardhat/test/e2e/periphery/controller-helper.ts

```diff
@@ -1158,7 +1116,9 @@ describe("ControllerHelper: mainnet fork", function () {
     const normFactor = await controller.getExpectedNormalizationFactor()
     const ethPrice = await oracle.getTwap(ethUsdcPool.address, weth.address, usdc.address, 420, true)
     const squeethPrice = await oracle.getTwap(wSqueethPool.address, wSqueeth.address, weth.address, 1, true)
-    const flashLoanAmount = (vaultBefore.shortAmount).mul(normFactor).mul(ethPrice).mul(3).div(one.mul(one).mul(10000).mul(2)).sub(vaultBefore.collateralAmount).add(ethers.utils.parseUnits('0.1'))
+    const flashLoanAmount_ = (vaultBefore.shortAmount).mul(normFactor).mul(ethPrice).mul(3).div(one.mul(one).mul(10000).mul(2)).sub(vaultBefore.collateralAmount).add(ethers.utils.parseUnits('0.1'))
+    const flashLoanAmount = flashLoanAmount_.lte(0) ? BigNumber.from(0) : flashLoanAmount_
+
```

Comment:
It was more just to be general so the tests would only take a flashloan if they needed it

Reply:
Yeah, I get that, but does what I'm saying make sense though?  

Say we have an issue with flashloan only trades in the code (not the edge case where flashloan amount = 0).  If we format the wSqueeth amounts and collateral amounts so the flashloan amount needed in negative, and thus gets set to 0, we won't test taking a flashloan.  And thus we could miss a base functionality test.  If the test reverts due to a -ve number to flashloan, we know that our initial conditions for the mint/collateral are wrong.

Reply:
Agree with Leone, if there is a bug that will only be found with a flashloaned amount, but this test set flashloan=0 because vault is well-overcollateralized, then we will never have a test case with flashloan amount. I like this generalized way, but I think tests should target a specific case.